src/main/scala/com/softwaremill/mqperf/Sender.scala

```diff
@@ -10,35 +11,28 @@ import scala.util.Random
 
 object Sender extends StrictLogging {
   def main(args: Array[String]): Unit = {
-    Thread.setDefaultUncaughtExceptionHandler((t, e) => { println("Uncaught exception in thread: " + t); e.printStackTrace() })
+    Thread.setDefaultUncaughtExceptionHandler((t, e) => {
+      println("Uncaught exception in thread: " + t); e.printStackTrace()
+    })
+
     import PrometheusMetricServer._
     withMetricsServerSync(CollectorRegistry.defaultRegistry) {
       logger.info("Starting sender...")
       val testConfig = TestConfig.load()
 
-      val mq = Mq.instantiate(testConfig)
-      val labelValues = defaultLabelValues(testConfig, TestConfig.hostId)
-      val c = Counter.build("mqperf_sent_total", "number of sent messages").labelNames(DefaultLabelNames: _*).register()
-      val g = Gauge
-        .build("mqperf_sent_threads_done", "number of sent threads done")
-        .labelNames(DefaultLabelNames: _*)
-        .register()
-      g.labels(labelValues: _*).set(0)
-      val h = Histogram
-        .build("mqperf_send_latency_ms", "latency of sent messages")
-        .buckets(0, 50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 600, 700, 800, 900, 1000, 1250, 1500, 1750, 2000,
-          2500, 3000, 3500, 4000, 4500, 5000)
-        .labelNames(DefaultLabelNames: _*)
-        .register()
+      val mq: Mq = Mq.instantiate(testConfig)
+      val (messagesCounter: Counter.Child, threadsDoneCounter: Gauge.Child, sendLatency: Histogram.Child) = createMetrics(testConfig)
+      val messagesPool: MessagesPool = MessagesPool.generatePoolOfRandomMessageOfLengthN(testConfig.msgSize, MessagesPool.DEFAULT_MESSAGES_POOL_SIZE)
```

Comment:
before we've been generating messages of length `msgSize - timestampLength`, so that after adding the timestamp you always go the reuqested length. I think this is missing here? That's why this used to be called `prefix` :)

Reply:
Thanks, updated. Technically speaking: we keep the length for any message of original length >= 13 characters. Otherwise it's just the length of the stringified timestamp.